sdk/storage/azure-storage-blob/tests/test_blob_tags_async.py

```diff
@@ -0,0 +1,294 @@
+        blob_client, resp = await self._create_block_blob()
+
+        # Act
+        blob_tags = {"tag1": "firsttag", "tag2": "secondtag", "tag3": "thirdtag"}
+        await blob_client.set_blob_tags(blob_tags)
+
+        resp = await blob_client.get_blob_tags()
+
+        # Assert
+        self.assertIsNotNone(resp)
+        self.assertEqual(len(resp), 3)
+        for key, value in resp.items():
+            self.assertEqual(blob_tags[key], value)
+
+    @GlobalStorageAccountPreparer()
+    @AsyncStorageTestCase.await_prepared_test
+    async def test_get_blob_tags_for_a_snapshot(self, resource_group, location, storage_account, storage_account_key):
+        await self._setup(storage_account, storage_account_key)
+        blob_tags = {"+-./:=_ ": "firsttag", "tag2": "+-./:=_", "+-./:=_1": "+-./:=_"}
+        blob_client, resp = await self._create_block_blob(blob_tags=blob_tags)
+
+        snapshot = await blob_client.create_snapshot()
+        snapshot_client = self.bsc.get_blob_client(self.container_name, blob_client.blob_name, snapshot=snapshot)
+
+        resp = await snapshot_client.get_blob_tags()
+
+        # Assert
+        self.assertIsNotNone(resp)
+        self.assertEqual(len(resp), 3)
+        for key, value in resp.items():
+            self.assertEqual(blob_tags[key], value)
+
+    @GlobalStorageAccountPreparer()
+    @AsyncStorageTestCase.await_prepared_test
+    async def test_upload_block_blob_with_tags(self, resource_group, location, storage_account, storage_account_key):
+        await self._setup(storage_account, storage_account_key)
+        blob_tags = {"tag1": "firsttag", "tag2": "secondtag", "tag3": "thirdtag"}
+        blob_client, resp = await self._create_block_blob(blob_tags=blob_tags)
+
+        resp = await blob_client.get_blob_tags()
+
+        # Assert
+        self.assertIsNotNone(resp)
+        self.assertEqual(len(resp), 3)
+
+    @GlobalStorageAccountPreparer()
+    @AsyncStorageTestCase.await_prepared_test
+    async def test_get_blob_properties_returns_tags_num(self, resource_group, location, storage_account, storage_account_key):
+        await self._setup(storage_account, storage_account_key)
+        blob_tags = {"tag1": "firsttag", "tag2": "secondtag", "tag3": "thirdtag"}
+        blob_client, resp = await self._create_block_blob(blob_tags=blob_tags)
+
+        resp = await blob_client.get_blob_properties()
+        downloaded = await blob_client.download_blob()
+
+        # Assert
+        self.assertIsNotNone(resp)
+        self.assertEqual(resp.tag_count, len(blob_tags))
+        self.assertEqual(downloaded.properties.tag_count, len(blob_tags))
+
+    @GlobalStorageAccountPreparer()
+    @AsyncStorageTestCase.await_prepared_test
+    async def test_create_append_blob_with_tags(self, resource_group, location, storage_account, storage_account_key):
+        await self._setup(storage_account, storage_account_key)
+        blob_tags = {"+-./:=_ ": "firsttag", "tag2": "+-./:=_", "+-./:=_1": "+-./:=_"}
```

Comment:
Do we need to test case sensitivity?

Reply:
It's supposed to be case sensitive, I updated the doc. While the service fix hasn't reached to the test tenant...So we cannot have an accurate client side test currently.